Scripts/Runtime/Entities/Tasks/AbstractTaskSystem.cs

```diff
@@ -117,5 +193,55 @@ private JobHandle UpdateTaskDriverSystem(JobHandle dependsOn)
         //         throw new InvalidOperationException($"Trying to add {taskDriver} to {this}'s list of Task Drivers but it is already there!");
         //     }
         // }
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        private void Debug_CheckFieldIsReadOnly(FieldInfo fieldInfo)
+        {
+            if (!fieldInfo.IsInitOnly)
+            {
+                throw new InvalidOperationException($"Field with name {fieldInfo.Name} on {fieldInfo.ReflectedType} is not marked as \"readonly\", please ensure that it is.");
+            }
+        }
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        private void Debug_CheckFieldTypeGenericTypeArguments(Type fieldType)
+        {
+            if (fieldType.GenericTypeArguments.Length != 1)
+            {
+                throw new InvalidOperationException($"Type {fieldType} is to be used to create a {typeof(ProxyDataStream<>)} but {fieldType} doesn't have the expected 1 generic type!");
+            }
+        }
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        private void Debug_CheckDataStreamUpdateJobsExist()
+        {
+            if (m_HasCheckedDataStreamUpdateJobsIntegrity)
+            {
+                return;
+            }
+
+            m_HasCheckedDataStreamUpdateJobsIntegrity = true;
```

Comment:
This will fail to compile when `ENABLE_UNITY_COLLECTIONS_CHECKS` isn't defined. Need to wrap in an `#if`

Reply:
Good catch, this is now changed as the bool is useful for other aspects of the state. It's no longer wrapped in `#if`